docs/en/ingest-management/elastic-agent/configuration/inputs/input-configuration.asciidoc

```diff
@@ -7,38 +7,70 @@
 
 The `inputs` section of the `elastic-agent.yml` file specifies how {agent} locates and processes input data.
 
-By default {agent} collects system metrics, such as CPU, memory, network, and file system metrics, and sends them to the default output. For example, to define the datastream for `cpu` metrics, this is the configuration:
+* <<elastic-agent-input-configuration-sample-metrics>>
+* <<elastic-agent-input-configuration-sample-logs>>
+
+[discrete]
+[[elastic-agent-input-configuration-sample-metrics]]
+== Sample metrics input configuration
+
+By default {agent} collects system metrics, such as CPU, memory, network, and file system metrics, and sends them to the default output. For example, to define datastreams for `cpu`, `memory`, `network` and `filesystem` metrics, this is the configuration:
 
 ["source","yaml"]
 -----------------------------------------------------------------------
-- id: unique-system-metrics-id <1>
-  type: system/metrics <2>
-  use_output: default <3>
-  meta:
-    package: <4>
-      name: system
-      version: 0.10.9
-  data_stream:
-    namespace: default <5>
+- type: system/metrics <1>
+  id: unique-system-metrics-id <2>
+  data_stream.namespace: default <3>
+  use_output: default <4>
   streams:
-    - data_stream:
-        dataset: system.cpu <6>
-        type: metrics <7>
-      metricsets: <8>
-        - cpu
-      period: 10s
-      cpu.metrics:
-        - percentages
-        - normalized_percentages
+    - metricsets: <5>
+      - cpu
+      data_stream.dataset: system.cpu <6>
+    - metricsets:
+      - memory
+      data_stream.dataset: system.memory
+    - metricsets:
+      - network
+      data_stream.dataset: system.network
+    - metricsets:
+      - filesystem
+      data_stream.dataset: system.filesystem
 -----------------------------------------------------------------------
 
-<1> A unique ID for the input.
-<2> A generic type describing the data.
-<3> The name of the `output` to use. If not specified, `default` will be used.
-<4> Package specification.
-<5> A user-defined namespace.
-<6> A user-defined dataset. It can contain anything that makes sense to signify the source of the data.
-<7> The type of the data stream.
-<8> Enabled module metricsets.
+<1> A generic type describing the data.
```

Comment:
This is not a generic type, this is the input type and needs to match one of the inputs supported by the Elastic-Agent. The supported input types are any Beat input or their aliases defined in the spec files, like [`filebeat.spec.yml`](https://github.com/elastic/elastic-agent/blob/698f7b99e5a26df9c2f311fec5ff792b4a11c4d6/specs/filebeat.spec.yml#L134-L137).

`data_stream.type` is a generic type describing the data that will be used to generate the index name.

The same applies to the other example down below.

Reply:
Nice! Thanks a lot for explaining this! To show what inputs are supported I made [these tables](https://ingest-docs_489.docs-preview.app.elstc.co/guide/en/fleet/master/elastic-agent-input-types.html). There were a few things I wasn't sure about though, in the third table (based on [filebeat.spec.yml](https://github.com/elastic/elastic-agent/blob/698f7b99e5a26df9c2f311fec5ff792b4a11c4d6/specs/filebeat.spec.yml)) and the fourth table (based on [heartbeat.spec.yml](https://github.com/elastic/elastic-agent/blob/698f7b99e5a26df9c2f311fec5ff792b4a11c4d6/specs/heartbeat.spec.yml)) so I left them blank with a `-`. If you have any suggestions for those that'd be great.

Reply:
Those tables look great @kilfoyle! I don't have much to add. I'm 95% sure the `docker` and `log/docker` are the `container` input.

I really like the idea of having an automated way to generate the docs and using the spec files looks great, they're a great source of truth for this.

Reply:
Thanks! I've changed `docker` and `log/docker` to be described as aliases for `container`. I'm glad you like the automation idea!